### PR TITLE
Add asMultipart method

### PR DESCRIPTION
**Pull request by faustbrian**

### CODE REVIEW

**Comment by adamwathan** [comment]
If you can add tests for this happy to merge.

File: src/Zttp.php [hunk 1]
```diff
@@ -54,6 +54,11 @@ function asFormParams()
         return $this->bodyFormat('form_params')->contentType('application/x-www-form-urlencoded');
     }
 
+    function asMultipart()
+    {
+        return $this->bodyFormat('multipart')->contentType("multipart/form-data; boundary=".md5(uniqid(null, true)));
```
Comment (adamwathan) [inline src/Zttp.php]: Is the `boundary` part (or even the content type actually, I realize I did that on the others but maybe not needed?) totally necessary? Guzzle won't take care of that with just the body format set?

Comment (faustbrian) [reply src/Zttp.php]: I tested it with multiple different HTTP Responses and sites like httpbin.org and the result is always a complain that the boundary is missing.

Comment (adamwathan) [reply src/Zttp.php]: This piece of code from Guzzle makes me think we can just not set the contentType stuff at all:

```php
$request = Psr7\modify_request($request, $modify);
if ($request->getBody() instanceof Psr7\MultipartStream) {
    // Use a multipart/form-data POST if a Content-Type is not set.
    $options['_conditional']['Content-Type'] = 'multipart/form-data; boundary='
        . $request->getBody()->getBoundary();
}
```

Could you test just removing that entire function call and seeing if things work as you expect?

Comment (faustbrian) [reply src/Zttp.php]: I removed the multipart content-type and it still works.

**Comment by faustbrian** [comment]
@adamwathan Could you take a look at this and merge it or close it so I can get rid of the repository? Thanks.

**Pull request merged**